evm-contracts/test/v0.6/BlockhashStore.test.ts

```diff
@@ -0,0 +1,132 @@
+      hash: "0x5cf096dfd1fc2d2947a96fdec5377ab7beaa0eb00c80728a3b96f0864cec506a",
+    },
+  ]
+
+  const deployment = setup.snapshot(provider, async () => {
+    blockhashStoreTestHelper = await blockhashStoreTestHelperFactory.connect(personas.Default).deploy()
+  })
+
+  beforeEach(async () => {
+    await deployment()
+
+    const [lastBlock] = mainnetBlocks.slice(-1);
+    await blockhashStoreTestHelper.connect(personas.Default)
+      .godmodeSetHash(lastBlock.num, lastBlock.hash)
+    assert.strictEqual(await blockhashStoreTestHelper.getBlockhash(lastBlock.num), lastBlock.hash)
+  })
+
+  it('getBlockhash reverts for unknown blockhashes', async () => {
+    matchers.evmRevert(
+      blockhashStoreTestHelper.getBlockhash(99999999),
+      "blockhash not found in store"
+    )
+  })
+
+  it('storeVerifyHeader records valid blockhashes', async () => {
+    for (let i = mainnetBlocks.length - 2; i >= 0; i--) {
+      await blockhashStoreTestHelper.connect(personas.Default)
+        .storeVerifyHeader(mainnetBlocks[i].num, mainnetBlocks[i + 1].rlpHeader)
+      assert.strictEqual(await blockhashStoreTestHelper.getBlockhash(mainnetBlocks[i].num), mainnetBlocks[i].hash)
+    }
+  })
+
+  it('storeVerifyHeader rejects unknown headers', async () => {
+    const [unknownBlock] = mainnetBlocks.slice(0, 1);
+    await matchers.evmRevert(
+      blockhashStoreTestHelper.connect(personas.Default)
+        .storeVerifyHeader(unknownBlock.num - 1, unknownBlock.rlpHeader),
+      "header has unknown blockhash"
+    )
+  })
+
+  it('storeVerifyHeader rejects corrupted headers', async () => {
+    const [lastBlock] = mainnetBlocks.slice(-1);
+    const modifiedHeader = new Uint8Array(lastBlock.rlpHeader)
+    modifiedHeader[137] += 1
+    await matchers.evmRevert(
+      blockhashStoreTestHelper.connect(personas.Default)
+        .storeVerifyHeader(lastBlock.num - 1, modifiedHeader),
+      "header has unknown blockhash"
+    )
+  })
+
+  it('store accepts recent block numbers', async () => {
+    await h.mineBlock(provider)
+
+    const n = await provider.getBlockNumber() - 1
+    await blockhashStoreTestHelper.connect(personas.Default).store(n)
+
+    await blockhashStoreTestHelper.getBlockhash(n)
```

Comment:
If it's easy to get the blockhash (it would be in golang), it's probably worth checking here that it's stored the right one.